### PR TITLE
Adds support for rails 8.0

### LINKED ISSUE #31: Add support for Rails 8
Ref: https://github.com/rails/rails/releases/tag/v8.0.0

### PR DESCRIPTION
**Pull request by manuelvanrijn**
* Rails 8.0 has been released, and this PR will bump the dependency and add an appraise for it.
* I've also removed the upper version constraint as mentiond by @ezekg in #31 
* Resolved some issues with the CI

[edgeguides.rubyonrails.org/8_0_release_notes.html](https://edgeguides.rubyonrails.org/8_0_release_notes.html)

Closes: #31

### CODE REVIEW

File: gemfiles/8.0.gemfile [hunk 1]
```diff
@@ -2,6 +2,6 @@
 
 source "https://rubygems.org"
 
-gem "activerecord", "~> 5.2"
+gem "activerecord", "~> 8.0"
```
Comment (manuelvanrijn) [inline gemfiles/8.0.gemfile:5]: 😆 weird it sees this as a move file :)

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -15,6 +15,10 @@ jobs:
           - '7.0'
           - '7.1'
           - '7.2'
+          - '8.0'
+        exclude:
```
Comment (ezekg) [inline .github/workflows/test.yml:19]: Assuming this is excluded because Rails changed the minimum Ruby version?

Can we make that clearer via a comment?

**Review by ezekg — CHANGES_REQUESTED** [review]

**Comment by ezekg** [comment]
Thanks a lot for the PR! One minor change i.r.t. CI.

**Comment by ezekg** [comment]
No worries, I got it. 👍 Published in v1.8.0!

**Comment by manuelvanrijn** [comment]
Thanks @ezekg and yes, it was because Rails 8 requires Ruby 3.2+ :)

**Pull request merged**